addon/factory.js

```diff
@@ -4,29 +4,21 @@ import _isFunction from 'lodash/lang/isFunction';
 import _isPlainObject from 'lodash/lang/isPlainObject';
 import _mapValues from 'lodash/object/mapValues';
 
+function buildValue(value, parentAttrs, sequence) {
+  if (_isArray(value)) {
+    return value.map((val) => buildValue(val, value, sequence));
+  } else if (_isPlainObject(value)) {
+    return _mapValues(value, (val) => buildValue(val, value, sequence));
+  } else if (_isFunction(value)) {
+    return value.call(parentAttrs, sequence);
+  } else {
+    return value;
+  }
+}
+
```

Comment:
I will have look what can be done tmrw, for now you can close the PR if you want